openapi3/openapi3.go

```diff
@@ -24,12 +25,48 @@ type T struct {
 	Tags         Tags                 `json:"tags,omitempty" yaml:"tags,omitempty"`
 	ExternalDocs *ExternalDocs        `json:"externalDocs,omitempty" yaml:"externalDocs,omitempty"`
 
+	// OpenAPI 3.1.x specific fields
+	// Webhooks are a new feature in OpenAPI 3.1 that allow APIs to define callback operations
+	Webhooks map[string]*PathItem `json:"webhooks,omitempty" yaml:"webhooks,omitempty"`
+
+	// JSONSchemaDialect allows specifying the default JSON Schema dialect for Schema Objects
+	// See https://spec.openapis.org/oas/v3.1.0#schema-object
+	JSONSchemaDialect string `json:"jsonSchemaDialect,omitempty" yaml:"jsonSchemaDialect,omitempty"`
+
 	visited visitedComponent
 	url     *url.URL
 }
 
 var _ jsonpointer.JSONPointable = (*T)(nil)
 
+// IsOpenAPI3_0 returns true if the document is OpenAPI 3.0.x
+func (doc *T) IsOpenAPI3_0() bool {
+	if doc == nil || doc.OpenAPI == "" {
+		return false
+	}
+	return doc.OpenAPI[0:3] == "3.0"
+}
+
+// IsOpenAPI3_1 returns true if the document is OpenAPI 3.1.x
+func (doc *T) IsOpenAPI3_1() bool {
+	if doc == nil || doc.OpenAPI == "" {
+		return false
+	}
+	return doc.OpenAPI[0:3] == "3.1"
```

Comment:
I think both `IsOpenAPI3_1` and `IsOpenAPI3_0` can be reduced to 
```suggestion
	return doc.Version() == "3.1"
```

since the `Version` function is already checking for `doc == nil || doc.OpenAPI == ""`